engine/src/hashchain.rs

```diff
@@ -0,0 +1,973 @@
+use self::blockchain_hashchain_error::BlockchainHashchainError;
+use crate::{bloom::Bloom, prelude::Vec};
+use aurora_engine_sdk::keccak;
+use aurora_engine_types::types::RawH256;
+
+#[cfg(not(feature = "borsh-compat"))]
+use borsh::{BorshDeserialize, BorshSerialize};
+#[cfg(feature = "borsh-compat")]
+use borsh_compat::{self as borsh, BorshDeserialize, BorshSerialize};
+
+/// Blockchain Hashchain.
+/// Continually keeps track of the previous block hashchain through the blocks heights.
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+pub struct BlockchainHashchain {
+    chain_id: [u8; 32],
+    contract_account_id: Vec<u8>,
```

Comment:
Why not `AccountId`?

Reply:
For the hashes, things are used in their bytes representation. We can use AccountId here, but I wanted to have as little dependencies as possible so hashchain code is easily replicated on other code bases or languages. For example, right now we are thinking on doing it on Go for the hashchain seed.

In any case, there is a good chance that we remove this field from the hashchain. It is nice to have and it should be easy to obtain, but it is not explicitly part of the Aurora blocks.

Reply:
We finally removed this one.